syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.qsn.api.rpc.grpc.proto.entity.request";

package org.qsn.protobuf;

import "qsn/entity/common/BaseRequest.proto";

message GetBlockHeaderByHashRequest {
    BaseRequest baseRequest = 1;
    bytes hash = 2;
}

message GetBlockHeaderByHeightRequest {
    BaseRequest baseRequest = 1;
    int64 height = 2;
}

message GetBlockByHashRequest {
    BaseRequest baseRequest = 1;
    bytes hash = 2;
}

message GetBlockByHeightRequest {
    BaseRequest baseRequest = 1;
    int64 height = 2;
}

message GetDoneHeightRequest {
    BaseRequest baseRequest = 1;
}

message GetBlockListRequest {
    BaseRequest baseRequest = 1;
    int64 startHeight = 2;
    int64 endHeight = 3;
}
